src/inventory/slot_mapping.rs

```diff
@@ -0,0 +1,196 @@
+use crate::inventory::{Item, Slot};
+use crate::render::hud::Hud;
+use crate::render::inventory::InventoryWindow;
+use crate::render::Renderer;
+use crate::ui::{Container, VAttach};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use parking_lot::RwLock;
+
+/// A mapping of each slot and it's position on the screen. This makes the job
+/// of rendering and locating slots easier for inventories. This class only
+/// expects positions in pixels relative to the top-left corner of the window.
+pub struct SlotMapping {
```

Comment:
we could try to make this internally mutable, as we would probably *always* employ locking from outside otherwise

Reply:
We could probably avoid locking if we keep using an HashMap(DashMap) and could now take an inventories' read lock to access the SlotMapping instead of having to take its write lock. Would that provide us with any benefits?

Reply:
I don't see too much value in doing this. At the moment it isn't modified too much. It seems to only be when a packet is sent from the server to update the slots, or when the user interacts with the window.